jcl/src/java.base/share/classes/java/security/AccessController.java

```diff
@@ -1027,13 +1027,10 @@ public static <T> T doPrivilegedWithCombiner(PrivilegedExceptionAction<T> action
 private static AccessControlContext doPrivilegedWithCombinerHelper(AccessControlContext context) {
 	ProtectionDomain domain = getCallerPD(2);
 	ProtectionDomain[] pdArray = (domain == null) ? null : new ProtectionDomain[] { domain };
-	AccessControlContext fixedContext = new AccessControlContext(context, pdArray, getNewAuthorizedState(context, domain));
-	if (context == null) {
-		AccessControlContext parentContext = getContextHelper(true);
-		fixedContext.domainCombiner = parentContext.domainCombiner;
-		fixedContext.nextStackAcc = parentContext;
-	}
-	return fixedContext;
+	AccessControlContext parentContext = getContextHelper(context == null);
+	DomainCombiner domaincombiner = parentContext.getCombiner();
+
+	return new AccessControlContext(pdArray, domaincombiner, parentContext, context, getNewAuthorizedState(context, domain));
```

Comment:
```suggestion
	return new AccessControlContext(pdArray, parentContext, context, getNewAuthorizedState(context, domain));
```
`domaincombiner = parentContext.getCombiner()` can be invoked within the constructor.

Reply:
It still passes both `parentContext.getCombiner()` and `parentContext`, only `parentContext` is needed.
In addition, pls use tabs instead of spaces to be consistent with the existing codebase.

Reply:
updated